govcd/landing_zone.go

```diff
@@ -0,0 +1,207 @@
+/*
+* Copyright 2024 VMware, Inc.  All rights reserved.  Licensed under the Apache v2 License.
+ */
+
+package govcd
+
+import (
+	"fmt"
+	"net/url"
+
+	"github.com/vmware/go-vcloud-director/v2/types/v56"
+)
+
+// SolutionLandingZone controls VCD Solution Add-On Landing Zone. It does so by wrapping RDE for
+// entity types vmware:solutions_organization:1.0.0.
+//
+// Up to VCD 10.5.1.1 ,there can only be one single RDE instance for landing zone.
+type SolutionLandingZone struct {
+	// SolutionLandingZoneType defines internal content of RDE (`types.DefinedEntity.State`)
+	SolutionLandingZoneType *types.SolutionLandingZoneType
+	// DefinedEntity contains parent defined entity that contains SolutionLandingZoneType in
+	// "Entity" field
+	DefinedEntity *DefinedEntity
+	vcdClient     *VCDClient
+}
+
+// CreateSolutionLandingZone configures VCD Solution Add-On Landing Zone. It does so by performing
+// the following steps:
+//
+// 1. Creates Solution Landing Zone RDE based on type urn:vcloud:type:vmware:solutions_organization:1.0.0
+// 2. Resolves the RDE
+func (vcdClient *VCDClient) CreateSolutionLandingZone(slzCfg *types.SolutionLandingZoneType) (*SolutionLandingZone, error) {
+	// 1. Check that RDE type exists
+	rdeType, err := vcdClient.GetRdeType("vmware", "solutions_organization", "1.0.0")
```

Comment:
Should we centralise this tuple `(vmware, solutions_organization, 1.0.0)` somewhere? As it is repeated across this file.
Maybe we could do
```
func getSolutionsLandingZoneRdeType() string, []string {
  rdeType := "vmware:solutions_organization:1.0.0"
  return rdeType, strings.Split(rdeType, ":")
}
```
(just an idea)

Reply:
Agreed. I had my doubts when putting this in, but lets do it if there is one more vote :)

Reply:
+1 :)

Reply:
Here is one approach - how does this one look? Would have been super cool if I could somehow expand it directly to function, but I think it doesn't work so.

Reply:
Works for me. Just this one slipped: https://github.com/vmware/go-vcloud-director/pull/665/files#diff-4482de8aee71583b72e4a7ac58ae72352f9903495e3205d67e4f305c05f567beR126

Reply:
Nice catch. Fixed.